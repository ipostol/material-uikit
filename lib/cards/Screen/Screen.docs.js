'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Screen = require('./Screen');

var _Screen2 = _interopRequireDefault(_Screen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    'div',
    { style: { height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center', backgroundColor: 'gray' } },
    _react2.default.createElement(_Screen2.default, {
      cost: '\xA34000',
      description: 'Description',
      places: 'Places',
      resolutions: 'Resolutions',
      images: ['mhnzlkatwgr8nt0j88th']
    })
  );
};