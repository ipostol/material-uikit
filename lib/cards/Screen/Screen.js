'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _bidstackCloudinary = require('bidstack-cloudinary');

var _bidstackCloudinary2 = _interopRequireDefault(_bidstackCloudinary);

var _Screen = require('./Screen.scss');

var _Screen2 = _interopRequireDefault(_Screen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Screen = function Screen(_ref) {
  var images = _ref.images,
      cost = _ref.cost,
      description = _ref.description,
      places = _ref.places,
      resolutions = _ref.resolutions,
      children = _ref.children,
      _ref$element = _ref.element,
      Element = _ref$element === undefined ? 'div' : _ref$element,
      rest = _objectWithoutProperties(_ref, ['images', 'cost', 'description', 'places', 'resolutions', 'children', 'element']);

  return _react2.default.createElement(
    Element,
    _extends({ className: (0, _classnames2.default)(_Screen2.default.general, rest.className) }, rest),
    children,
    images.length > 1 ? _react2.default.createElement(
      'div',
      { className: _Screen2.default.imagesWrap },
      _react2.default.createElement('img', {
        src: _bidstackCloudinary2.default.url(images[0], { crop: 'fill', width: 200, height: 180 }),
        alt: '',
        className: _Screen2.default.firstImage
      }),
      _react2.default.createElement(
        'div',
        { className: _Screen2.default.wrap },
        _react2.default.createElement('img', {
          src: _bidstackCloudinary2.default.url(images[1], { crop: 'fill', width: 117, height: 88 }),
          alt: '',
          className: _Screen2.default.image
        }),
        _react2.default.createElement('img', {
          src: _bidstackCloudinary2.default.url(images[2], { crop: 'fill', width: 117, height: 88 }),
          alt: '',
          className: _Screen2.default.image
        })
      )
    ) : _react2.default.createElement('img', {
      src: _bidstackCloudinary2.default.url(images[0], { crop: 'fill', width: 337, height: 211 }),
      alt: '',
      className: _Screen2.default.mainImage
    }),
    _react2.default.createElement(
      'div',
      { className: _Screen2.default.infoWrap },
      _react2.default.createElement(
        'div',
        { className: _Screen2.default.cost },
        cost
      ),
      _react2.default.createElement(
        'div',
        { className: _Screen2.default.screens },
        description
      ),
      _react2.default.createElement(
        'div',
        { className: _Screen2.default.caption },
        places
      ),
      _react2.default.createElement(
        'div',
        { className: _Screen2.default.caption },
        resolutions
      )
    )
  );
};

Screen.propTypes = {
  images: _propTypes2.default.array,
  cost: _propTypes2.default.string,
  description: _propTypes2.default.string,
  places: _propTypes2.default.string,
  resolutions: _propTypes2.default.string,
  children: _propTypes2.default.node,
  element: _propTypes2.default.any
};

exports.default = Screen;