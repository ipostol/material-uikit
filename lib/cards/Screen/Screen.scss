@import '~bidstack-theme/theme';

.general {
  width: 337px;
  border-radius: 4px;
  background-color: $white3;
  margin-bottom: 16px;
  text-decoration: none;
  color: $fontColorMain;
}

.infoWrap {
  margin: 7px 16px 19px;
}

.cost {
  font-size: $sizeBodyM;
  line-height: 1.47;
  margin-bottom: 7px;
}

.screens {
  font-size: $sizeCaption;
  line-height: 1.23;
  margin-bottom: 2px;
}

.caption {
  font-size: $sizeCaption;
  line-height: 1.23;
  color: $fontColorLight2;
}

.imagesWrap {
  margin: 0 8px;
  display: flex;
  flex-shrink: 0;
}

.wrap {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.firstImage {
  border-radius: 2px;
}

.image {
  margin-left: 4px;
  border-radius: 2px;
}

.mainImage {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}
