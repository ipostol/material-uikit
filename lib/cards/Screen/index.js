'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _Screen = require('./Screen');

var _Screen2 = _interopRequireDefault(_Screen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _Screen2.default;