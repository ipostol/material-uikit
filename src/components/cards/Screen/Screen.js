import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import cloudinary from 'bidstack-cloudinary';
import styles from './Screen.scss';

const Screen = ({ images, cost, description, places, resolutions, children, element: Element = 'div', ...rest }) => (
  <Element className={cn(styles.general, rest.className)} {...rest}>
    {children}
    {images.length > 1 ? <div className={styles.imagesWrap}>
      <img
        src={cloudinary.url(images[0], { crop: 'fill', width: 200, height: 180 })}
        alt=""
        className={styles.firstImage}
      />
      <div className={styles.wrap}>
        <img
          src={cloudinary.url(images[1], { crop: 'fill', width: 117, height: 88 })}
          alt=""
          className={styles.image}
        />
        <img
          src={cloudinary.url(images[2], { crop: 'fill', width: 117, height: 88 })}
          alt=""
          className={styles.image}
        />
      </div>
    </div>
  : <img
    src={cloudinary.url(images[0], { crop: 'fill', width: 337, height: 211 })}
    alt=""
    className={styles.mainImage}
  />}
    <div className={styles.infoWrap}>
      <div className={styles.cost}>{cost}</div>
      <div className={styles.screens}>{description}</div>
      <div className={styles.caption}>{places}</div>
      <div className={styles.caption}>{resolutions}</div>
    </div>
  </Element>
);

Screen.propTypes = {
  images: PropTypes.array,
  cost: PropTypes.string,
  description: PropTypes.string,
  places: PropTypes.string,
  resolutions: PropTypes.string,
  children: PropTypes.node,
  element: PropTypes.any,
};

export default Screen;
