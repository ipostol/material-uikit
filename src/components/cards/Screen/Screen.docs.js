import React from 'react';
import Screen from './Screen';

export default () => (
  <div style={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center', backgroundColor: 'gray' }}>
    <Screen
      cost="Â£4000"
      description="Description"
      places="Places"
      resolutions="Resolutions"
      images={['mhnzlkatwgr8nt0j88th']}
    />
  </div>
);
