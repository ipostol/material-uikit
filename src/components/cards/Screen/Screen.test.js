import React from 'react';
import { shallow } from 'enzyme';
import Screen from './Screen';

describe('Screen component', () => {

  it('should render correctly', () => {

    const props = {
      images: ['img'],
    };

    const wrapper = shallow(
      <Screen {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly many images', () => {

    const props = {
      images: ['img', 'img', 'img'],
    };

    const wrapper = shallow(
      <Screen {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
