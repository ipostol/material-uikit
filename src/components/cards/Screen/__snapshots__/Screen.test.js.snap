// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen component should render correctly 1`] = `
<div
  className="general"
>
  <img
    alt=""
    className="mainImage"
    src="http://res.cloudinary.com/bidstack-img/image/upload/c_fill,h_211,w_337/img"
  />
  <div
    className="infoWrap"
  >
    <div
      className="cost"
    />
    <div
      className="screens"
    />
    <div
      className="caption"
    />
    <div
      className="caption"
    />
  </div>
</div>
`;

exports[`Screen component should render correctly many images 1`] = `
<div
  className="general"
>
  <div
    className="imagesWrap"
  >
    <img
      alt=""
      className="firstImage"
      src="http://res.cloudinary.com/bidstack-img/image/upload/c_fill,h_180,w_200/img"
    />
    <div
      className="wrap"
    >
      <img
        alt=""
        className="image"
        src="http://res.cloudinary.com/bidstack-img/image/upload/c_fill,h_88,w_117/img"
      />
      <img
        alt=""
        className="image"
        src="http://res.cloudinary.com/bidstack-img/image/upload/c_fill,h_88,w_117/img"
      />
    </div>
  </div>
  <div
    className="infoWrap"
  >
    <div
      className="cost"
    />
    <div
      className="screens"
    />
    <div
      className="caption"
    />
    <div
      className="caption"
    />
  </div>
</div>
`;
